### PR TITLE
Animations: apply calc/var transforms to keyframes and timing

### PR DESCRIPTION
**Pull request by dvoytenko**
Partial for #9129.

### CODE REVIEW

File: extensions/amp-animation/0.1/web-animations.js [hunk 1]
```diff
@@ -634,14 +647,262 @@ export class MeasureScanner extends Scanner {
...
+    this.baseUrl_ = baseUrl;
+
+    /** @private {!Object<string, !CSSStyleDeclaration>} */
+    this.computedStyleCache_ = map();
+
+    /** @private {!Object<string, ?./css-expr-ast.CssNode>} */
+    this.parsedCssCache_ = map();
+
+    /** @private {?Element} */
+    this.current_ = null;
```
Comment (aghassemi) [inline extensions/amp-animation/0.1/web-animations.js]: nit: `currentTarget_`

File: extensions/amp-animation/0.1/css-expr-ast.js [hunk 2]
```diff
@@ -17,6 +17,20 @@
 const FINAL_URL_RE = /^(data|https)\:/i;
 const DEG_TO_RAD = 2 * Math.PI / 360;
 const GRAD_TO_RAD = Math.PI / 200;
+const VAR_CSS_RE = /calc|var|url/i;
```
Comment (dvoytenko) [inline extensions/amp-animation/0.1/css-expr-ast.js]: Done. Though false positives here are very benign.

File: examples/animations.amp.html [hunk 3]
```diff
@@ -38,23 +47,25 @@
...
       }
     },
     {
       "selector": "#image2",
       "easing": "cubic-bezier(0,0,.21,1)",
-      "keyframes": {
-        "transform": "translateX(0px)"
-      }
+      "keyframes": [
+        {"transform": "translateY(var(--start-y))", "opacity": "var(--start-o)"},
```
Comment (dvoytenko) [inline examples/animations.amp.html]: Ack!

File: extensions/amp-animation/0.1/web-animations.js (hunk 1 above)
Comment (dvoytenko) [inline extensions/amp-animation/0.1/web-animations.js]: Done

**Pull request merged**